cache/async_cache.go

```diff
@@ -20,6 +20,8 @@ type AsyncCache struct {
 	TransactionRegistry
 
 	graceTime time.Duration
+
+	MaxPayloadSize int64
```

Comment:
Yes, ByteSize is a custom type based on uint64. In the Cache struct maxPayloadSize is a ByteSize, also in the config defaultMaxPayloadSize is a ByteSize. The reason why in the AsyncCache struct it's int64 because we need compare it with the contentLength/bufferedRespWriter which is int64. There is a function which checks if the contentLength not greater than maxPayloadSize, and if the MaxPayloadSize is ByteSize we will need to convert it to int64 every time  we check the contentLength

Reply:
still, I agree with @mga-chka we should not allow building a cache with negative value